x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/PushDownAndCombineLimits.java

```diff
@@ -64,7 +65,7 @@ public LogicalPlan rule(Limit limit, LogicalOptimizerContext ctx) {
                     }
                 }
             }
-        } else if (limit.child() instanceof Join join && join.config().type() == JoinTypes.LEFT) {
+        } else if (limit.child() instanceof Join join && join.config().type() == JoinTypes.LEFT && join instanceof InlineJoin == false) {
```

Comment:
++.

Maybe let's add a comment that we _want_ to also apply this to InlineJoin, but that requires that the stub relation corresponds to a node somewhere _inside_ the left branch, rather than the whole left branch. Or we need to get rid of the stub relation then and replace it by a copy of the left branch (with updated name ids, most likely, but that's a detail).